Editor/AGS.Editor/Resources/agsdefns.sh

```diff
@@ -457,6 +457,60 @@ internalstring autoptr builtin managed struct String {
   readonly import attribute int Length;
 };
 
+#ifdef SCRIPT_API_v350
+builtin managed struct Dictionary
+{
+  /// Creates a new empty Dictionary of the given properties.
+  import static Dictionary* Create(bool sorted = false, bool caseSensitive = false); // $AUTOCOMPLETESTATICONLY$
+
+  /// Removes all items from the dictionary.
+  import void Clear();
+  /// Tells if given key is in the dictionary.
+  import bool Contains(String key);
+  /// Gets value by the key; returns null if such key does not exist.
+  import String Get(String key);
+  /// Removes key/value pair from the dictionary, fails if there was no such key.
+  import bool Remove(String key);
+  /// Assigns a value to the given key, adds this key if it did not exist yet.
+  import bool Set(String key, String value);
+
+  /// Gets if this dictionary is case-sensitive.
+  import readonly attribute bool CaseSensitive;
+  /// Gets if this dictionary is sorted by keys in alphabetical order.
+  import readonly attribute bool Sorted;
+  /// Gets the number of key/value pairs currently in the dictionary.
+  import readonly attribute int ItemCount;
+  /// Accesses the keys in the dictionary, sorted or unsorted depending on the dictionary's property.
+  import readonly attribute String Keys[];
+  /// Accesses the values in the dictionary, stored in the order of their keys.
+  import readonly attribute String Values[];
+};
+
+builtin managed struct Set
+{
+  /// Creates a new empty Set of the given properties.
+  import static Set* Create(bool sorted = false, bool caseSensitive = false); // $AUTOCOMPLETESTATICONLY$
+
+  /// Adds item to the set, fails if such item was already existing.
+  import bool Add(String item);
+  /// Removes all items from the set.
+  import void Clear();
+  /// Tells if given item is in the set.
+  import bool Contains(String item);
+  /// Removes item from the set, fails if there was no such item.
+  import bool Remove(String item);
+
+  /// Gets if this set is case-sensitive.
+  import readonly attribute bool CaseSensitive;
+  /// Gets if this set is sorted in alphabetical order.
+  import readonly attribute bool Sorted;
+  /// Gets the number of items currently in the set.
+  import readonly attribute int ItemCount;
+  /// Accesses the items in the set, sorted or unsorted depending on the set's property.
+  import readonly attribute String Items[];

```

Comment:
Might be worth nothing if modifying the Set affects this getting or not. I still wonder if it's better to just have a "ToSortedArray" method instead.

Reply:
I have doubts about returning an array *instead* (as opposed to complementary method): the most common use for iterating over set/dictionary content is to display it somewhere, and if a person would like to have it done repeatedly and keep synced at the same time I'm afraid they will do it in a very suboptimal way (creating new array every game tick). Not every ags user knows these nuances.

But I can agree that this Items[] property seem a bit off for this type of container. Maybe there's a way to make a proper "iterator" which only lets to pass container from start to finish in a direct sequence.